lib/srv/db/sqlserver/connect.go

```diff
@@ -93,12 +126,16 @@ func (c *connector) Connect(ctx context.Context, sessionCtx *common.Session, log
 			return nil, nil, trace.Wrap(err)
 		}
 	} else {
-		auth, err := c.getAuth(sessionCtx)
+		kc, err := c.getKerberosClient(ctx, sessionCtx)
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+		a, err := c.getAuth(sessionCtx, kc)
```

Comment:
nit:
even if the  scope of 'a' variable is really narrow  the `auth` var name looks better: 
```suggestion
		auth, err := c.getAuth(sessionCtx, kc)
		 ...
	         connector = mssql.NewConnectorConfig(dsnConfig, auth)
```

Reply:
`dbAuth` sounds good